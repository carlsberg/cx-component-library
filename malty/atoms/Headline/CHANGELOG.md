# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0 (2024-05-31)

### Bug Fixes

- **DSM-521:** update children type ([#532](https://github.com/CarlsbergGBS/cx-component-library/issues/532)) ([152012c](https://github.com/CarlsbergGBS/cx-component-library/commit/152012cff67b1c4ef6d0df119bd343d9c8f40f5d))
- Remove typography provider from every component ([#383](https://github.com/CarlsbergGBS/cx-component-library/issues/383)) ([5d05680](https://github.com/CarlsbergGBS/cx-component-library/commit/5d05680e40d3da5bb3042bd8191427a70606afc3))
- reverted changes ([#396](https://github.com/CarlsbergGBS/cx-component-library/issues/396)) ([ecfcf57](https://github.com/CarlsbergGBS/cx-component-library/commit/ecfcf573863253fea16e24764b6f452c3cc4f215))
- Run auto fixable lint based on new `eslint` configuration ([#501](https://github.com/CarlsbergGBS/cx-component-library/issues/501)) ([9ee94cc](https://github.com/CarlsbergGBS/cx-component-library/commit/9ee94cc2cd407f717e62d8857b5f20a74e7bbec4))

### Features

- add tsconfig files and update package.json files for each component ([3bfc644](https://github.com/CarlsbergGBS/cx-component-library/commit/3bfc644e1cfc9dbb7cf7a0469e25fce055b53240))
- change organization name ([a0aa428](https://github.com/CarlsbergGBS/cx-component-library/commit/a0aa428f43138af5707a3ff4c0e36b7c056f02e6))
- **DSM-228:** Added rating component ([#379](https://github.com/CarlsbergGBS/cx-component-library/issues/379)) ([ab47354](https://github.com/CarlsbergGBS/cx-component-library/commit/ab47354db803fb051a1f87d66ee34966698df8bd))
- **DSM-248:** Headline HTML tags ([#184](https://github.com/CarlsbergGBS/cx-component-library/issues/184)) ([305f5e0](https://github.com/CarlsbergGBS/cx-component-library/commit/305f5e0fa92d5813f4877b338ad827a8c347d452))
- **DSM-362:** snapshots removed & unit test updates ([#307](https://github.com/CarlsbergGBS/cx-component-library/issues/307)) ([78423eb](https://github.com/CarlsbergGBS/cx-component-library/commit/78423eb5cfe04fe8a3a7af1f5af074ac6d35e235))
- **DSM-439:** Change font-family behavior ([#470](https://github.com/CarlsbergGBS/cx-component-library/issues/470)) ([ab6dad4](https://github.com/CarlsbergGBS/cx-component-library/commit/ab6dad414c42c4343de3de6abaa4cd3ff98cfca1))
- **DSM-495:** Fixed build warnings/errors ([#499](https://github.com/CarlsbergGBS/cx-component-library/issues/499)) ([051bc5a](https://github.com/CarlsbergGBS/cx-component-library/commit/051bc5ac296d7117995d970efce64523f31d4918))
- **DSM-495:** Update Headline as prop ([#491](https://github.com/CarlsbergGBS/cx-component-library/issues/491)) ([5a14155](https://github.com/CarlsbergGBS/cx-component-library/commit/5a1415505cc8598d2af7d7ffcb27aa7bbc4717be))
- **DSM-516:** Article Card ([#521](https://github.com/CarlsbergGBS/cx-component-library/issues/521)) ([0ba55f9](https://github.com/CarlsbergGBS/cx-component-library/commit/0ba55f923bcdd37a63ff3aef77ff0e42046e777e))
- **DSM-551:** Add word break to Headline ([#550](https://github.com/CarlsbergGBS/cx-component-library/issues/550)) ([a8a6616](https://github.com/CarlsbergGBS/cx-component-library/commit/a8a66168e8ac794f911a8f69d5d70cefe20df2d1))
- **global:** [DSM-784] add space props in all components ([#842](https://github.com/CarlsbergGBS/cx-component-library/issues/842)) ([9f03f59](https://github.com/CarlsbergGBS/cx-component-library/commit/9f03f59e7518a99abcea06bacbee989d1ddb0465))
- **global:** migrate from bit to lerna ([a0110c8](https://github.com/CarlsbergGBS/cx-component-library/commit/a0110c8831370dc762c193b17cc593eed381f990))
- **global:** update storybook files to be compliant with v7 ([#970](https://github.com/CarlsbergGBS/cx-component-library/issues/970)) ([c61d4b4](https://github.com/CarlsbergGBS/cx-component-library/commit/c61d4b40a0755becf942ad3f28758a159f8c54e4))
- **headline:** [DSM-834] add tests ([#802](https://github.com/CarlsbergGBS/cx-component-library/issues/802)) ([0f575f6](https://github.com/CarlsbergGBS/cx-component-library/commit/0f575f6df41377284d4a41e0b3edaa2d0e9cc5e6))
- **hero, banner:** [DSM-620] Create new Hero and banner component ([#650](https://github.com/CarlsbergGBS/cx-component-library/issues/650)) ([28ca296](https://github.com/CarlsbergGBS/cx-component-library/commit/28ca29674647143244183e45539f1e60b25d8c0b))
- remove empty composition files ([f8173f4](https://github.com/CarlsbergGBS/cx-component-library/commit/f8173f4a2ecbf80bb7b6ffe848c023ae31819c2d))
- Several component updates ([#185](https://github.com/CarlsbergGBS/cx-component-library/issues/185)) ([633c9f0](https://github.com/CarlsbergGBS/cx-component-library/commit/633c9f0a4432e9e3d08ff05156453660929c19a2))
- Use new `eslint-config-carlsberg` with streamlined way to create components ([#559](https://github.com/CarlsbergGBS/cx-component-library/issues/559)) ([a459332](https://github.com/CarlsbergGBS/cx-component-library/commit/a45933215bc0b523220743f17d988d7f1ad5fce6))
- v1.0.29 minor updates ([#247](https://github.com/CarlsbergGBS/cx-component-library/issues/247)) ([73a6675](https://github.com/CarlsbergGBS/cx-component-library/commit/73a6675abfa3ff73cb0bb334d6e2cd63e6198e88))
