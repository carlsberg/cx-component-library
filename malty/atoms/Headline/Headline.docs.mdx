---
labels: ['react', 'typescript', 'headline']
description: 'A Headline component.'
---
