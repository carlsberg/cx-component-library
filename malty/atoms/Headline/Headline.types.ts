import { SpaceProps } from '@carlsberggroup/malty.utils.space';
import React, { ReactNode } from 'react';

export interface HeadlineProps extends React.HTMLAttributes<HTMLHeadingElement>, SpaceProps {
  headlineStyle?: HeadlineStyle;
  align?: HeadlineAlign;
  color?: HeadlineColor;
  as?: React.ElementType;
  children: ReactNode;
  dataTestId?: string;
}

export enum HeadlineStyle {
  Display = 'display',
  Banner = 'banner',
  Huge = 'huge',
  Big = 'big',
  Large = 'large',
  MediumLarge = 'medium-large',
  Medium = 'medium'
}
export enum HeadlineAlign {
  Left = 'left',
  Center = 'center',
  Right = 'right'
}
export enum HeadlineColor {
  ThemePrimary = 'theme-primary',
  DigitalBlack = 'digital-black',
  White = 'white'
}

export type StyledHeadlineProps = {
  headlineStyle: HeadlineStyle;
  color: HeadlineColor;
  align: HeadlineAlign;
} & SpaceProps;
