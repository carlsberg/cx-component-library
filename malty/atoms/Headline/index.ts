export { Headline } from './Headline';
export { StyledHeadline } from './Headline.styled';
export { Align, Color, Size } from './Headline.types';
export type { HeadlineProps } from './Headline.types';
