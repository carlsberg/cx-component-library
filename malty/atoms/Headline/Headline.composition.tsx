import React from 'react';
import { Headline } from './Headline';

export const BasicText = () => <Headline>Headline</Headline>;
