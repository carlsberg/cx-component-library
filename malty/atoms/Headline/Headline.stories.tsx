import { generateStorybookSpacing } from '@carlsberggroup/malty.utils.space';
import { Story } from '@storybook/react';
import React from 'react';
import { Headline as HeadlineComponent } from './Headline';
import { HeadlineAlign, HeadlineColor, HeadlineProps, HeadlineStyle } from './Headline.types';

export default {
  title: 'Typography/Headline',
  component: HeadlineComponent,
  parameters: {
    importObject: 'Headline',
    importPath: '@carlsberggroup/malty.atoms.headline',
    themed: true
  },
  argTypes: {
    color: {
      description: 'Color of the text, from the following options',
      options: Object.keys(HeadlineColor),
      mapping: HeadlineColor,
      control: {
        type: 'select',
        label: Object.values(HeadlineColor)
      },
      table: {
        defaultValue: {
          summary: 'HeadlineColor.DigitalBlack'
        }
      },
      defaultValue: 'DigitalBlack'
    },
    headlineStyle: {
      description: 'Size of the text, from the following options',
      control: {
        type: null
      },
      defaultValue: 'Medium'
    },
    align: {
      description: 'Text alignment, from the following options',
      options: Object.keys(HeadlineAlign),
      mapping: HeadlineAlign,
      control: {
        type: 'select',
        label: Object.values(HeadlineAlign)
      },
      table: {
        defaultValue: {
          summary: 'HeadlineAlign.Left'
        }
      },
      defaultValue: 'Left'
    },
    children: {
      description: 'This is the content of a headline component',
      control: { type: 'text' }
    },
    as: {
      description: "HTML tag override to be used, from 'h1' through 'h6', as well as 'p' or 'span' tags.",
      control: { type: 'text' }
    },
    ...generateStorybookSpacing()
  }
};
const Template: Story<HeadlineProps> = (args) => (
  <>
    <HeadlineComponent {...args} headlineStyle={HeadlineStyle.Display} />
    <HeadlineComponent {...args} headlineStyle={HeadlineStyle.Banner} />
    <HeadlineComponent {...args} headlineStyle={HeadlineStyle.Huge} />
    <HeadlineComponent {...args} headlineStyle={HeadlineStyle.Big} />
    <HeadlineComponent {...args} headlineStyle={HeadlineStyle.Large} />
    <HeadlineComponent {...args} headlineStyle={HeadlineStyle.MediumLarge} />
    <HeadlineComponent {...args} headlineStyle={HeadlineStyle.Medium} />
  </>
);
export const Headline = Template.bind({});

Headline.args = {
  children: 'This is a sample headline'
};
