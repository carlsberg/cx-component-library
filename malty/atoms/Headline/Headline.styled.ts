import styled, { css } from 'styled-components';
import { HeadlineAlign, HeadlineColor, HeadlineStyle } from './Headline.types';

export const StyledHeadline = styled.h1<{
  headlineStyle: HeadlineStyle;
  color: HeadlineColor;
  align: HeadlineAlign;
}>`
  margin: 0;
  padding: 0;
  color: ${({ color, theme }) => {
    let tempColor;
    if (color === HeadlineColor.ThemePrimary) {
      tempColor = theme.colors.theme.themePrimary.value;
    } else {
      tempColor = theme.colors['headline-colours'][color].value;
    }
    return tempColor;
  }};
  ${({ theme, headlineStyle }) =>
    css`
      font-family: ${theme.typography.desktop.headline[headlineStyle]['font-family'].value};
      font-size: ${theme.typography.desktop.headline[headlineStyle]['font-size'].value};
      letter-spacing: ${theme.typography.desktop.headline[headlineStyle]['letter-spacing'].value};
      line-height: ${theme.typography.desktop.headline[headlineStyle]['line-height'].value};
      font-weight: ${theme.typography.desktop.headline[headlineStyle]['font-weight'].value};
    `}
  text-align: ${({ align }) => align ?? HeadlineAlign.Left};
  ${({ theme, headlineStyle }) =>
    theme.typography.desktop.headline[headlineStyle]['text-transform']?.value &&
    css`
      text-transform: ${theme.typography.desktop.headline[headlineStyle]['text-transform']?.value};
    `}
  ${({ theme, headlineStyle }) =>
    theme.typography.desktop.headline[headlineStyle]['text-decoration']?.value &&
    css`
      text-decoration: ${theme.typography.desktop.headline[headlineStyle]['text-decoration']?.value};
    `}
  ${({ headlineStyle, theme }) => css`
    @media screen and (max-width: ${theme.layout.small['device-max-width']?.value}) {
      ${css`
        font-family: ${theme.typography.tablet.headline[headlineStyle]['font-family'].value};
        font-size: ${theme.typography.tablet.headline[headlineStyle]['font-size'].value};
        letter-spacing: ${theme.typography.tablet.headline[headlineStyle]['letter-spacing'].value};
        line-height: ${theme.typography.tablet.headline[headlineStyle]['line-height'].value};
        font-weight: ${theme.typography.tablet.headline[headlineStyle]['font-weight'].value};
      `}
      ${theme.typography.tablet.headline[headlineStyle]['text-transform']?.value &&
      css`
        text-transform: ${theme.typography.tablet.headline[headlineStyle]['text-transform']?.value};
      `}
    ${theme.typography.tablet.headline[headlineStyle]['text-decoration']?.value &&
      css`
        text-decoration: ${theme.typography.tablet.headline[headlineStyle]['text-decoration']?.value};
      `}
    }
    @media screen and (max-width: ${theme.layout.xsmall['device-max-width']?.value}) {
      ${css`
        font-family: ${theme.typography.mobile.headline[headlineStyle]['font-family'].value};
        font-size: ${theme.typography.mobile.headline[headlineStyle]['font-size'].value};
        letter-spacing: ${theme.typography.mobile.headline[headlineStyle]['letter-spacing'].value};
        line-height: ${theme.typography.mobile.headline[headlineStyle]['line-height'].value};
        font-weight: ${theme.typography.mobile.headline[headlineStyle]['font-weight'].value};
      `}
      ${theme.typography.mobile.headline[headlineStyle]['text-transform']?.value &&
      css`
        text-transform: ${theme.typography.mobile.headline[headlineStyle]['text-transform']?.value};
      `}
    ${theme.typography.mobile.headline[headlineStyle]['text-decoration']?.value &&
      css`
        text-decoration: ${theme.typography.mobile.headline[headlineStyle]['text-decoration']?.value};
      `}
    }
  `}
  word-break: normal;
  overflow-wrap: anywhere;
`;
