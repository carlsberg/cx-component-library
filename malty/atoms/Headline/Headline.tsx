import { globalTheme as defaultTheme } from '@carlsberggroup/malty.theme.malty-theme-provider';
import React, { useContext } from 'react';
import { ThemeContext } from 'styled-components';
import { StyledHeadline } from './Headline.styled';
import { Align, Color, HeadlineProps, Size } from './Headline.types';

export const Headline = ({
  size = Size.Medium,
  align = Align.Left,
  color = Color.Primary,
  children
}: HeadlineProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;

  return (
    <StyledHeadline size={size} align={align} color={color} theme={theme}>
      {children}
    </StyledHeadline>
  );
};
