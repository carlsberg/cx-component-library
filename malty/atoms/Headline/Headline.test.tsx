import { jsonRenderer, render, screen } from '@carlsberggroup/malty.utils.test';
import React from 'react';
import { Headline } from './Headline';

const text = 'this is a headline';
describe('Headline component', () => {
  it('matches snapshot', () => {
    const view = jsonRenderer(<Headline>{text}</Headline>);
    expect(view).toMatchSnapshot();
  });

  it('should render with the correct text', () => {
    render(<Headline>{text}</Headline>);
    const rendered = screen.getByText(text);
    expect(rendered).toBeInTheDocument();
  });
});
