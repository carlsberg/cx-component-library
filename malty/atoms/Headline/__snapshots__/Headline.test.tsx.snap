// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline component matches snapshot 1`] = `
<p
  className="sc-bdnxRM iLZrOY"
>
  this is a headline
</p>
`;
